server/aesdsocket.c

```diff
@@ -110,56 +130,98 @@ void listen_for_connections(int sockfd) {
 }
 
 // Handles client connections
-void handle_client(int client_fd) {
-    FILE *fp = fopen(CUSTOM_LOG_FILE, "a+");
+void *handle_client(void *arg) {
+    int client_fd = *((int *)arg);
+    free(arg);
+    
+    add_thread(pthread_self());
+    
     char buffer[MAX_CUSTOM_BUFFER];
+    char packetBuffer[MAX_CUSTOM_BUFFER];  // Buffer to accumulate a complete packet
+    size_t packetSize = 0;
 
     while (1) {
         ssize_t bytes_recv = recv(client_fd, buffer, sizeof(buffer), 0);
         if (bytes_recv <= 0) {
             break;
         }
 
-        fwrite(buffer, bytes_recv, 1, fp);
+        // Accumulate data until a complete packet is received
+        memcpy(packetBuffer + packetSize, buffer, bytes_recv);
+        packetSize += bytes_recv;
 
+        // Check if a complete packet (ending with '\n') is received
         if (memchr(buffer, '\n', bytes_recv) != NULL) {
+            // Write the complete packet to the log file using system call
+            pthread_mutex_lock(&list_mutex);
+            int fd = open(CUSTOM_LOG_FILE, O_RDWR | O_CREAT | O_APPEND, 0777);
+            if (fd == -1) {
+                syslog(LOG_INFO, "Couldn't open file");
+                pthread_exit(NULL);
+            }
+            if (write(fd, packetBuffer, packetSize) == -1) {
+		syslog(LOG_INFO, "Couldn't write to file");
+		exit(EXIT_FAILURE);
+	    }
+            
+            // Read from the log file and send back to the client
+	    lseek(fd, 0, SEEK_SET);  // Set the file pointer to the beginning of the file
+
+	    while (1) {
+		ssize_t bytes_read = read(fd, buffer, sizeof(buffer));
+		if (bytes_read <= 0) {
+		    break;
+		}
+
+		// Send the contents of the log file back to the client
+		send(client_fd, buffer, bytes_read, 0);
+	    }
+
+            close(fd);
+            pthread_mutex_unlock(&list_mutex);
+            
+            // Reset the packet buffer for the next packet
+            memset(packetBuffer, 0, sizeof(packetBuffer));
+            packetSize = 0;
+
+            // Break after handling the complete packet
             break;
         }
     }
 
-    fclose(fp);
-
-    fp = fopen(CUSTOM_LOG_FILE, "r");
-    while (!feof(fp)) {
-        ssize_t bytes_read = fread(buffer, 1, sizeof(buffer), fp);
-        if (bytes_read <= 0) {
-            break;
-        }
-
-        send(client_fd, buffer, bytes_read, 0);
-    }
-
-    fclose(fp);
     close(client_fd);
+    remove_thread(pthread_self());
+    pthread_exit(NULL);
 }
 
 // Accepts incoming client connections
 void accept_clients(int sockfd) {
-    while (1) {
+    while (!sig_exit) {
         struct sockaddr_in client_addr;
         socklen_t client_addr_size = sizeof(client_addr);
-        int client_fd = accept(sockfd, (struct sockaddr *)&client_addr, &client_addr_size);
+        int *client_fd = malloc(sizeof(int));
```

Comment:
@Suhas-Reddy-S shouldn't need to malloc this or use a pointer, just use `int client_fd` and use `&` when you need to pass the address to `pthread_create`